ocaml/runtime/major_gc.c

```diff
@@ -289,12 +289,6 @@ void caml_darken (value v, value *p)
       h = Hd_val (v);
       t = Tag_hd (h);
     }
-#ifdef NO_NAKED_POINTERS
-    /* We insist that naked pointers to outside the heap point to things that
-       look like values with headers coloured black.  This is always
-       strictly necessary because the compactor relies on it. */
-    CAMLassert (Is_in_heap (v) || Is_black_hd (h));
-#endif
```

Comment:
If the page table is populated anyway, these assertions could check it directly via classify_addr?

Reply:
As I discussed with @stedolan offline, this isn't entirely straightforward due to the new structure of the code in `caml_page_table_lookup`, so we'll leave this.  The NNP checker is better for this anyway.